nsxt/resource_nsxt_policy_tier0_inter_vrf_routing.go

```diff
@@ -0,0 +1,389 @@
+				inFilter = interface2StringList(brlMap["out_filter"].([]interface{}))
+			}
+			bgpRouteLeaking = append(bgpRouteLeaking, model.BgpRouteLeaking{
+				AddressFamily: &addressFamily,
+				InFilter:      inFilter,
+				OutFilter:     outFilter,
+			})
+		}
+	}
+
+	var staticRouteAdvertisement *model.PolicyStaticRouteAdvertisement
+	staticRouteAdvert := d.Get("static_route_advertisement")
+	if staticRouteAdvert != nil {
+		for _, sAdvert := range staticRouteAdvert.([]interface{}) {
+			// Should be one as list has MaxItems = 1
+			sAdvertMap := sAdvert.(map[string]interface{})
+
+			var advertisementRules []model.PolicyRouteAdvertisementRule
+			advertRulesList := sAdvertMap["advertisement_rule"]
+			if advertRulesList != nil {
+				for _, advRule := range advertRulesList.([]interface{}) {
+					advRuleMap := advRule.(map[string]interface{})
+					action := advRuleMap["action"].(string)
+					name := advRuleMap["name"].(string)
+					prefixOperator := advRuleMap["prefix_operator"].(string)
+
+					var routeAdvertisementTypes []string
+					if advRuleMap["route_advertisement_types"] != nil {
+						routeAdvertisementTypes = interface2StringList(advRuleMap["route_advertisement_types"].([]interface{}))
+					}
+
+					var subnets []string
+					if advRuleMap["subnets"] != nil {
+						subnets = interface2StringList(advRuleMap["subnets"].([]interface{}))
+					}
+
+					advertisementRules = append(advertisementRules, model.PolicyRouteAdvertisementRule{
+						Action:                  &action,
+						Name:                    &name,
+						PrefixOperator:          &prefixOperator,
+						RouteAdvertisementTypes: routeAdvertisementTypes,
+						Subnets:                 subnets,
+					})
+				}
+			}
+			var inFilterPrefixList []string
+			if sAdvertMap["in_filter_prefix_list"] != nil {
+				inFilterPrefixList = interface2StringList(sAdvertMap["in_filter_prefix_list"].([]interface{}))
+			}
+			staticRouteAdvertisement = &model.PolicyStaticRouteAdvertisement{
+				AdvertisementRules: advertisementRules,
+				InFilterPrefixList: inFilterPrefixList,
+			}
+		}
+	}
+
+	targetPath := d.Get("target_path").(string)
+
+	return model.PolicyInterVrfRoutingConfig{
+		DisplayName:              &displayName,
+		Description:              &description,
+		Tags:                     tags,
+		BgpRouteLeaking:          bgpRouteLeaking,
+		StaticRouteAdvertisement: staticRouteAdvertisement,
+		TargetPath:               &targetPath,
+	}
+}
+
+func resourceNsxtPolicyTier0InterVRFRoutingCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	client := tier_0s.NewInterVrfRoutingClient(connector)
+
+	gwPolicyPath := d.Get("gateway_path").(string)
+	isT0, gwID := parseGatewayPolicyPath(gwPolicyPath)
+	if gwID == "" {
+		return fmt.Errorf("gateway_path is not valid")
+	}
+	if !isT0 {
+		return fmt.Errorf("Tier0 gateway path expected, got %s", gwPolicyPath)
+	}
+
+	id := d.Get("nsx_id").(string)
+	if id == "" {
+		id = newUUID()
+	}
+
```

Comment:
We usually check that object with this `nsx_id` does not exist yet, here this check is missing, is this intentional?

Reply:
No, I guess that I've copy-pasted from the wrong place :)
Seems that there are quite a lot of cases where we do not check this. Maybe it's worthwhile to add this to some TODO list?